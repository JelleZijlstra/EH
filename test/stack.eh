#!/usr/bin/ehi
class Stack
  private n = 0
  private s

  public initialize = () => (this.s = [])
  
  public push = func: input
    this.s->n = input
    this.n = this.n + 1
  end
  
  public pop = func:
    printvar this.s
    if this.n == 0
      throw "Stack empty"
    end
    this.n = this.n - 1
    ret this.s->(this.n)
  end
  
  public size = () => this.n
end

s = Stack.new ()
printvar s.size ()
s.push 42
printvar s.size ()
printvar s.pop ()

printvar s.pop ()
