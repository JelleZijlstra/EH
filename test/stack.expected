@int 0
@int 1
@array [
0 => @int 42
]
@int 42
@array [
0 => @int 42
]
Uncaught exception of type String: Stack empty
